### DISCUSSION TITLE
The restore functions should be using long for the TTL not int

### DISCUSSION THREAD

**Discussion by NeutrinoAPI** [xetorthio/jedis#2373] [General]
Hi there,

The restore* group of functions use int as the "ttl" parameter, I believe this should actually be long because the restore command sets the TTL as milliseconds NOT seconds. 

As per the redis docs: https://redis.io/commands/restore
"If ttl is 0 the key is created without any expire, otherwise the specified expire time (in milliseconds) is set".

This leads to integer overflows when passing PTTLs with sufficiently large size...
I think this should be filed as a bug in jedis but adding as a discussion first in case I have missed something about this feature..

Cheers!

**Comment by sazzad16** [comment]
Hi @NeutrinoAPI, could you please create an issue for this? Thanks.

**Reply by NeutrinoAPI** [reply]
Done!

**Comment by yangbodong22011** [comment]
closed by https://github.com/redis/jedis/pull/2693
